src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Net46.cs

```diff
@@ -0,0 +1,1193 @@
+            SetConnectionOptions(webRequest, request);
+            SetServicePointOptions(webRequest, request);
+            SetRequestHeaders(webRequest, request);
+            SetContentHeaders(webRequest, request);
+
+            // New properties for this OOB HttpClientHandler.
+            webRequest.ServicePoint.ConnectionLimit = _maxConnectionsPerServer;
+            webRequest.MaximumResponseHeadersLength = _maxResponseHeadersLength;
+            if ((ClientCertificateOptions == ClientCertificateOption.Manual)
+                && (_clientCertificates != null) && (_clientCertificates.Count > 0))
+            {
+                webRequest.ClientCertificates = _clientCertificates;
+            }
+
+            if (_serverCertificateCustomValidationCallback != null)
+            {
+                webRequest.ServerCertificateValidationCallback = ServerCertificateValidationCallback;
+            }
+
+            if (_defaultProxyCredentials != null && webRequest.Proxy != null)
+            {
+                webRequest.Proxy.Credentials = _defaultProxyCredentials;
+            }
+
+            // For Extensibility
+            InitializeWebRequest(request, webRequest);
+
+            return webRequest;
+        }
+
+        // Used to map the ServerCertificateCustomValidationCallback which uses Func<T> to the
+        // HttpWebRequest based RemoteCertificateValidationCallback delegate type.
+        private bool ServerCertificateValidationCallback(
+            object sender,
+            X509Certificate certificate,
+            X509Chain chain,
+            SslPolicyErrors sslPolicyErrors)
+        {
+            return _serverCertificateCustomValidationCallback(
+                null, // TODO Issue #11100: How to map sender (which is an HttpWebRequest) to an HttpRequestMessage?
+                (X509Certificate2)certificate, // This cast will usually always be safe.
+                chain,
+                sslPolicyErrors);
+        }
+
+        // Needs to be internal so that WebRequestHandler can access it from a different assembly.
+        internal virtual void InitializeWebRequest(HttpRequestMessage request, HttpWebRequest webRequest)
+        {
+        }
+
+        private void SetDefaultOptions(HttpWebRequest webRequest)
+        {
+            webRequest.Timeout = Timeout.Infinite; // Timeouts are handled by HttpClient.
+
+            webRequest.AllowAutoRedirect = _allowAutoRedirect;
+            webRequest.AutomaticDecompression = _automaticDecompression;
+            webRequest.PreAuthenticate = _preAuthenticate;
+
+            if (_useDefaultCredentials)
+            {
+                webRequest.UseDefaultCredentials = true;
+            }
+            else
+            {
+                webRequest.Credentials = _credentials;
+            }
+
+            if (_allowAutoRedirect)
+            {
+                webRequest.MaximumAutomaticRedirections = _maxAutomaticRedirections;
+            }
```

Comment:
This is the code from Desktop.  The behavior needs to be exact as-is with original .NET Framework code.